File: AMS_基础信息组件/code/server/source/com.yss.ams.base.information.support/src/main/java/com/yss/ams/base/information/support/sys/ieItem/service/IIeItemDataService.java
package com.yss.ams.base.information.support.sys.ieItem.service;

import java.util.List;

import com.yss.ams.base.information.support.sys.ieItem.pojo.IeItem;
import com.yss.framework.api.cache.IDataServiceForCache;
import com.yss.framework.api.common.co.BasePojo;
import com.yss.framework.api.dataservice.GenericPojo;
import com.yss.framework.api.dataservice.IControlDataService;
import com.yss.framework.api.dataservice.IKeyConvertDataService;
import com.yss.framework.api.restful.annotations.RestfulSupported;
import com.yss.framework.api.service.ServiceException;

//import dataservice.comm.pojo.IeItem;

/**
 * 收支项目字典数据服务接口，主要进行跨应用数据获取
 * 字典表：T_S_IE_ITEM
 *
 */
@RestfulSupported
@GenericPojo(pojo = IeItem.class)
public interface IIeItemDataService extends IControlDataService,IKeyConvertDataService,IDataServiceForCache {

	/**
	 * 根据收支项目类型获取收支项目
	 * @param types 收支项目类型
	 * @return
	 * @throws ServiceException
	 * @author liuxiang 2015-8-14 STORY #24240 太平资产收支结转功能优化
	 */
	public <K extends BasePojo> List<K> getDataListByIeTypes(String[] types)
			throws ServiceException;
}

File: AMS_基础信息组件/code/client/source/YssInformation.Support/Sys/Dictionary/Service/IIeItemDataService.cs

using System.Collections.Generic;
using FAST.Common.Service.DataService.Base;
using FAST.Common.Service.Pojo.Base;
using FAST.Common.Service.DataService;
using FAST.Common.Service.Services.Base;
using FAST.Common.Service.Attributes;

////namespace YssBaseCls.Service
////namespace YssInformation.Sys.Dictionary.Service
namespace YssInformation.Support.Sys.Dictionary.Service
{
    /// <summary>
    /// 收支项目字典数据服务接口
    /// 
    /// Created By XiaoZhilong 2013-4-26
    /// </summary>
    [NewServiceAttribute(ServiceType.NEW_SVC)]
    [ServiceAttribute(ServiceId = YssInformation.Support.common.ServiceIdConstant.OSGI_BASEBUSINESS, ServiceController = "com.yss.ams.base.information.support.sys.ieItem.controller.IIeItemDataController", ServiceUrl = "")]
    public interface IIeItemDataService : IControlDataService, IKeyConvertDataService
    {
        /// <summary>
        /// 根据收支项目类型获取收支项目
        /// liuxiang 2015-8-14 STORY #24240 太平资产收支结转功能优化
        /// </summary>
        /// <param name="types">收支项目类型</param>
        /// <returns>收支项目列表</returns>
        [MethodAttribute(Method = RequestType.POST, MethodUrl = "/getDataListByIeTypes")]
        List<BasePojo> getDataListByIeTypes(string[] types);
    }
}
